### PR TITLE
Added heap size control and fixed a bug

### PR DESCRIPTION
**Pull request by ywkim312**

### CODE REVIEW

File: charts/geoserver/values.yaml [hunk 1]
```diff
@@ -44,6 +44,10 @@ persistence:
 # the values should be only url without http:// or https://, like "geoserver.example.com"
 whitelist: ""
 
+# the following is for the geoserver java memory heap size control
+envs:
+  extraJavaOpts: "-Xms512m -Xmx2g"
```
Comment (robkooper) [inline charts/geoserver/values.yaml:49]:
```suggestion
  EXTRA_JAVA_OPTS: "-Xms512m -Xmx2g"
```

File: charts/geoserver/templates/deployment.yaml [hunk 2]
```diff
@@ -40,10 +40,12 @@ spec:
...
               value: {{ .Values.cors.enabled | quote }}
-            - name: SKIP_DEMO_DATA
-              values: {{ .Values.demoData.skip | quote }}
             - name: GEOSERVER_CSRF_WHITELIST
-              values: {{ .Values.geoserver.whitelist | quote }}
+              value: {{ .Values.whitelist | quote }}
+            - name: SKIP_DEMO_DATA
+              value: {{ .Values.demoData.skip | quote }}
+            - name: EXTRA_JAVA_OPTS
+              value: {{ .Values.envs.extraJavaOpts | quote }}
```
Comment (robkooper) [inline charts/geoserver/templates/deployment.yaml:48]:
```suggestion
{{ with .Values.envs }}
            - name: .key
               value: .value
{{ end }}
```

File: charts/geoserver/README.md [hunk 3]
```diff
@@ -91,6 +91,11 @@ $ helm install --set persistence.existingClaim=PVC_NAME geoserver
 
 ## ChangeLog
 
+### 1.1.0
+
+- Added extra configuration options
```
Comment (robkooper) [inline charts/geoserver/README.md:96]: make sure to make more clear
also mention you changed http -> geoserver

**Review by robkooper — CHANGES_REQUESTED** [review]
update changelog, ignore the changes to `env` for now, make it an issue.

File: charts/geoserver/templates/deployment.yaml (hunk 2 above)
Comment (ywkim312) [reply charts/geoserver/templates/deployment.yaml:48]: Create an issue about this in https://github.com/ncsa/charts/issues/17

File: charts/geoserver/README.md (hunk 3 above)
Comment (ywkim312) [reply charts/geoserver/README.md]: updated chagelog

File: charts/geoserver/values.yaml (hunk 1 above)
Comment (ywkim312) [reply charts/geoserver/values.yaml:49]: made an issue about this in https://github.com/ncsa/charts/issues/17

**Pull request merged**